src/gl/gl_builders.js

```diff
@@ -145,213 +195,99 @@ GLBuilders.buildPolylines = function (
         closed_polygon,
         remove_tile_edges,
         texcoord_index,
-        texcoord_scale
+        texcoord_scale,
+        scaling_index
     }) {
 
-    // Build triangles
+    // Build variables
     var vertices = [],
-        texcoords = [],
-        p,
-        pa,
-        pb,
+        scalingVecs = scaling_index && [],   //  vertices directions for then (on GLSL vertex shader) extrude
+        texcoords = texcoord_index && [],
+        halfWidth = width/2,
         num_lines = lines.length;
-
+        
     var [[min_u, min_v], [max_u, max_v]] = texcoord_scale || [[0, 0], [1, 1]];
 
+    var constants = { vertex_data, vertex_template, halfWidth, vertices, scaling_index, scalingVecs, texcoord_index, texcoords, min_u, min_v, max_u, max_v };
```

Comment:
Maybe add a comment like:
`// Values that are constant for each line and are passed to helper functions`